cms/server/contest/handlers/tasksubmission.py

```diff
@@ -579,6 +579,7 @@ class UseTokenHandler(BaseHandler):
     """Called when the user try to use a token on a submission.
 
     """
+    # TODO: Decide if tokens can be applied after contest ends.
```

Comment:
Since tokens are sometimes used to determine the score, I'd say that it's reasonable to *not* allow users to apply tokens after the contest has ended.

Furthermore, when tokens are used to decide the score, playing a token is a something like sending a submission (i.e. if you play a token when it's too late, you won't get points, and that's how it's meant to be).